src/Uno.UI/UI/Xaml/Controls/Layouter/Layouter.cs

```diff
@@ -653,54 +684,34 @@ private Rect ApplyMarginAndAlignments(View view, Rect frame)
 		}
 
 		private double GetActualSize(
-			double size,
+			double frameSize,
 			bool isStretch,
-			double childMaxHeight,
-			double childMinHeight,
-			double childHeight,
-			double childMargins,
-			bool hasChildHeight,
-			bool hasChildMaxHeight,
-			bool hasChildMinHeight,
-			double desiredSize,
-			double frameSize)
+			double childMaxSize,
+			double childMinSize,
+			double childSize,
+			double childMarginSize,
+			bool hasChildSize,
+			bool hasChildMaxSize,
+			bool hasChildMinSize,
+			double desiredSize)
 		{
-			//Default value
-			//childVerticalAlignment != VerticalAlignment.Stretch
-			var actualHeight = Min(size, desiredSize);
-
-			if (hasChildHeight)
-			{
-				actualHeight = Min(childHeight + childMargins, size);
-			}
-			else if (hasChildMaxHeight && hasChildMinHeight)
-			{
-				actualHeight = Min(childMaxHeight + childMargins,
-					isStretch
-						? size
-						: desiredSize
-				);
-
-				actualHeight = Max(childMinHeight + childMargins, actualHeight);
-			}
-			else if (hasChildMaxHeight)
+			var min = hasChildMinSize ? childMinSize + childMarginSize : NegativeInfinity;
+			var max = hasChildMaxSize ? childMaxSize + childMarginSize : PositiveInfinity;
+			if (!hasChildSize)
 			{
-				actualHeight = Min(childMaxHeight + childMargins,
-					isStretch
-						? size
-						: desiredSize
-				);
+				childSize = isStretch
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Introduce a new variable instead of reusing the parameter 'childSize'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4826507)